src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/FeignInterceptorConfiguration.java

```diff
@@ -31,6 +38,9 @@ public RequestInterceptor requestInterceptor(FeignHeaderConfig config) {
                         String name = headerNames.nextElement();
                         String value = request.getHeader(name);
                         if (config.getHeaders().contains(name.toLowerCase())) {
+                            if (name.equalsIgnoreCase(AUTHORISATION)) {
+                                value = s2sClient.signIntoS2S();
```

Comment:
dont need to create new class /functionality for S2S . Just try below line and see if it works works.
authTokenGenerator will be instantiated with PRD S2S credentials from vault when sprint boot starts.

value  = authTokenGenerator.generate();

Reply:
The changes have been made in this commit: https://github.com/hmcts/rd-professional-api/pull/910/commits/c30848e3bc7dbe2db43a99dae26ffd39a2ecc9cf